x/mint/keeper/keeper_test.go

```diff
@@ -719,3 +719,348 @@ func (suite *KeeperTestSuite) TestDistributeDeveloperRewards() {
 		})
 	}
 }
+
+// TestCalculateTruncationsDelta tests that the per-epoch truncation delta is calculated
+// correctly. That includes checking that the pre-existing delta is
+// extracted from the store at the desired index and summed with the new delta
+// from the input values.
+func (suite *KeeperTestSuite) TestCalculateTruncationDelta() {
+	const mintDenom = sdk.DefaultBondDenom
+	tests := map[string]struct {
+		preExistingStoreDelta sdk.Dec
+
+		moduleAccountName string
+		provisions        sdk.Dec
+		amountDistributed sdk.Int
+
+		expectedTruncationDelta sdk.Dec
+
+		expectError bool
+	}{
+		"100.6 - 100 = 0.6; no pre-existing delta; inflation delta module account": {
+			moduleAccountName: types.DeveloperVestingModuleAcctName,
+			provisions:        sdk.NewDecWithPrec(1006, 1),
+			amountDistributed: sdk.NewInt(100),
+
+			expectedTruncationDelta: sdk.NewDecWithPrec(6, 1),
+		},
+		"100.6 - 100  + 0.3 = 0.9; pre-existing delta of 0.3; inflation delta module account": {
+			preExistingStoreDelta: sdk.NewDecWithPrec(3, 1),
+
+			moduleAccountName: types.ModuleName,
+			provisions:        sdk.NewDecWithPrec(1006, 1),
+			amountDistributed: sdk.NewInt(100),
+
+			expectedTruncationDelta: sdk.NewDecWithPrec(9, 1),
+		},
+		"100.6 - 100  + 0.5 = 1.1; pre-existing delta of 1.1; inflation delta module account": {
+			preExistingStoreDelta: sdk.NewDecWithPrec(5, 1),
+
+			moduleAccountName: types.ModuleName,
+			provisions:        sdk.NewDecWithPrec(1006, 1),
+			amountDistributed: sdk.NewInt(100),
+
+			expectedTruncationDelta: sdk.NewDecWithPrec(11, 1),
+		},
+		"82191.178 - 17808  + 33.33 = 64416508; pre-existing delta of 33.33; developer vesting module account": {
+			preExistingStoreDelta: sdk.NewDecWithPrec(3333, 2),
+
+			moduleAccountName: types.DeveloperVestingModuleAcctName,
+			provisions:        sdk.NewDecWithPrec(82191178, 3),
+			amountDistributed: sdk.NewInt(17808),
+
+			expectedTruncationDelta: sdk.NewDecWithPrec(64416508, 3),
+		},
+		"attempt to use invalid module account name - panic": {
+			moduleAccountName: poolincentivestypes.ModuleName,
+			provisions:        sdk.NewDecWithPrec(1006, 1),
+			amountDistributed: sdk.NewInt(100),
+
+			expectError: true,
+		},
+		"invalid module account - error": {
+			moduleAccountName: poolincentivestypes.ModuleName,
+			provisions:        sdk.NewDecWithPrec(1006, 1),
+			amountDistributed: sdk.NewInt(100),
+
+			expectError: true,
```

Comment:
Maybe not for this PR, but in general I feel like we should start moving away from bool errors and instead specify the error. Lmk what you think